overlord/snapstate/snapstate.go

```diff
@@ -36,7 +36,7 @@ import (
 	"github.com/snapcore/snapd/store"
 )
 
-func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup) (*state.TaskSet, error) {
+func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, maybeCore bool) (*state.TaskSet, error) {
```

Comment:
I'm not a big fan of a boolean argument like `maybeCore` because for the caller its hard to read what the boolean means: `doInstall(st, snapst, snapsup, true)` is not immediately clear what "true" or "false" mean here. Maybe a named flag or something? But not a blocker, just nitpicking.